imgui_plot.h

```diff
@@ -1088,6 +1072,55 @@ namespace ImGui
         ImPieProperties properties_;
     };
 
+    class ImAreaProperties : public ImLineProperties, public ImBarProperties
+    {
+    };
+
+    template<typename Ta, typename Tb>
+    class ImArea
+    {
+    public:
+        void DrawArea(
+            ImPlot<Ta, Tb>& figure,
+            const Ta* x_data,
+            const Tb* y_data,
+            const size_t n_data,
+            float * const y_data_bottoms = NULL
+        )
+        {
+            if (n_data < 2) return;
+
+            ImGuiWindow* window = GetCurrentWindow();
+            if (window->SkipItems)
+                return;
+
+            if (y_data_bottoms) {
+                for (size_t i = 0; i < n_data; ++i) {
+                    y_data_bottoms[i] += y_data[i];
+                }
+            }
+            float const * const y = y_data_bottoms ? y_data_bottoms : y_data;
+
+            ImScales<Ta, float>* scalesX = figure.GetScalesX();
```

Comment:
Let's try to avoid using intermediate points of these objects, and instead just use the entire chain (i.e., figure.GetScalesX().Scale(...)) similar to what I was doing in the other methods.